llvm/lib/Target/AIE/AIE2InstrInfo.cpp

```diff
@@ -52,6 +52,20 @@ using namespace llvm;
 #include "AIE2GenPreSchedLowering.inc"
 #include "AIE2GenSplitInstrTables.inc"
 
+// Return true if TRC is a superclass of RC or contains the given reg.
+// This is primarily a helper function for the functions below.  The first
+// case is active when Reg is a virtual register, but is apparently not
+// sufficient alone
+static bool regClassMatches(const TargetRegisterClass &TRC,
+                            const TargetRegisterClass *RC, unsigned Reg) {
+  if (RC && TRC.hasSubClassEq(RC))
+    return true;
+  if (TRC.contains(Reg))
+    return true;
+  return false;
+}
+
+#include "AIE2GenVarInstructionItin.inc"
```

Comment:
using already defined `regClassMatches` in `AIE2GenVarInstructionItin.inc`

Reply:
That is a weird dependency. Maybe we can make `regClassMatches` into a static method of `AIEBaseRegisterInfo`? Then `AIE2GenVarInstructionItin.inc` can just use `AIEBaseRegisterInfo::regClassMatches(...)`? This is also better for code sharing between AIE versions